pkg/controllers/job/job_controller.go

```diff
@@ -327,6 +359,37 @@ func (cc *jobcontroller) processNextReq(count uint32) bool {
 
 	klog.V(3).Infof("Try to handle request <%v>", req)
 
+	// Handle cancellation logic for delayed actions:
+	// 1. When request event is not Pod Pending:
+	//   - Cancel corresponding Pod Pending delayed action
+	//   - When Pod enters Running state, cancel corresponding Pod Failed and Pod Evicted delayed actions
+	if req.Event != busv1alpha1.PodPendingEvent {
+		cc.delayActionMapLock.Lock()
+		if taskMap, exists := cc.delayActionMap[key]; exists {
+			for podName, delayAct := range taskMap {
+				shouldCancel := false
+
+				if podName == req.PodName {
```

Comment:
why use `for key := range taskMap if key == xxx` rather  than `val, exists := taskMap[key]`?

Reply:
updated